### ISSUE TITLE
Upgrading from 0.33.1 to 0.33.2 causes ImageBuild timeout 

### ISSUE THREAD

**Issue by saweber** [tilt-dev/tilt#6177]
### Expected Behavior
[This sample repository] should successfully build and deploy my example application on v0.33.2

### Current Behavior
Attempting to build the container times out.

```STEP 1/5 — Building Dockerfile: [example-go-image-tilt_docker_build_with_restart_base]
Building Dockerfile for platform linux/arm64:
  FROM golang:alpine as dlv-env
  
  RUN go install github.com/go-delve/delve/cmd/dlv@latest
  
  FROM alpine:latest
  
  WORKDIR /app
  
  COPY --from=dlv-env /go/bin/dlv /go/bin/dlv
  
  ADD main main
  
  EXPOSE 8000 40000
  ENTRYPOINT ["/go/bin/dlv", "--listen=:40000", "--headless=true", "--api-version=2", "--accept-multiclient", "--log", "exec", "/app/main"]

     Building image
     
     ERROR IN: [internal] load .dockerignore
     
     ERROR IN: [internal] load build definition from Dockerfile

Build Failed: ImageBuild: failed to read dockerfile: no active session for k5b7eah03710pr8w1qql0dpbb: context deadline exceeded
```

### Steps to Reproduce

1. Run `tilt up` in the root of this sample repository: [This sample repository](https://github.com/saweber/go-tilt-hello-world)
2. With 0.33.2, I am seeing the above error.
3. Downgrading to 0.33.1 resolves the issue.
4. Building and running the container in docker with: `docker build -t example-go-image:test .` and `docker run -p 8000:8000 -p 40000:40000 example-go-image:test` also work.

#### `tilt doctor` Output

```
> tilt doctor
Tilt: v0.33.2, built 2023-07-19
System: darwin-arm64
---
Docker
- Host: unix:///Users/stevenweber/.docker/run/docker.sock
- Server Version: 24.0.2
- API Version: 1.43
- Builder: 2
- Compose Version: v2.19.1
---
Kubernetes
- Env: kind
- Context: kind-kind
- Cluster Name: kind-kind
- Namespace: default
- Container Runtime: containerd
- Version: v1.27.3
- Cluster Local Registry: none
---
Thanks for seeing the Tilt Doctor!
Please send the info above when filing bug reports. 💗

The info below helps us understand how you're using Tilt so we can improve,
but is not required to ask for help.
---
Analytics Settings
--> (These results reflect your personal opt in/out status and may be overridden by an `analytics_settings` call in your Tiltfile)
- User Mode: opt-out
- Machine: 789b29f94bf2310233b09cd621f1c7fe
- Repo: UDp5AvOsaaVL0is7l7buTA==
```

#### About Your Use Case

At the moment, forces me to downgrade to the previous version.

If there is any other context I can provide, please let me know! I'm running latest on docker/k8s/kind, as well as other dev dependencies. Hardware is a M2 Macbook Air.

Thank you!

**Comment by nicks** [comment]
hmmm thanks for the report. I haven't been able to reproduce this problem. Are you able to reproduce this with the sample projects like https://github.com/tilt-dev/tilt-example-html, or does it only happen with this project?

**Comment by chase-miller** [comment]
I’m having the same issue. Not at a computer to provide details, but it’s precisely the same.

**Comment by zacharyarnaise** [comment]
I can confirm this. I was able to reproduce the issue with all examples of https://github.com/tilt-dev/tilt-example-html and the sample repository provided by @saweber
Here my `tilt doctor` output:
```sh
Tilt: v0.33.2, built 2023-07-19
System: linux-amd64
---
Docker
- Host: unix:///var/run/docker.sock
- Server Version: 24.0.4
- API Version: 1.43
- Builder: 2
- Compose Version: v2.19.1
---
Kubernetes
- Env: k3d
- Context: k3d-k3s-default
- Cluster Name: k3d-k3s-default
- Namespace: default
- Container Runtime: containerd
- Version: v1.25.7+k3s1
- Cluster Local Registry: &RegistryHosting{Host:localhost:32795,HostFromClusterNetwork:registry:5000,HostFromContainerRuntime:registry:5000,Help:https://k3d.io/usage/guides/registries/#using-a-local-registry,SingleName:,}
---
Thanks for seeing the Tilt Doctor!
Please send the info above when filing bug reports. 💗
```

**Comment by nicks** [comment]
ah, i think i know what the problem is. i can repro if i compile with Go 1.20.6, which has some breaking changes. We're working with the Go team on getting them rolled back, see https://github.com/golang/go/issues/61431

Can you check that the version of Tilt you have is compiled with Go 1.20.6? You can check with something like:

```
strings ~/Downloads/tilt | grep go1.
```

Do you know where you're getting your `tilt` binaries from? The official release binaries on the releases page (https://github.com/tilt-dev/tilt/releases/tag/v0.33.2) are pinned to Go 1.20.5, AFAICT.

**Comment by zacharyarnaise** [comment]
Good catch, the version of Tilt 0.33.2 I got has been compiled with Go 1.20.6
Checked Tilt 0.33.1 and this is one has been compiled with Go 1.20.5

I'm getting the binaries through Homebrew: https://github.com/Homebrew/homebrew-core/blob/master/Formula/tilt.rb

**Comment by chase-miller** [comment]
Homebrew for me as well

**Comment by nicks** [comment]
The simplest fix right now is to get the binaries from the tilt tap instead of homebrew-core

```
brew install tilt-dev/tap/tilt
```

That will use the recompiled binaries. There might be a way to get homebrew-core to compile tilt with a pinned go version, but haven't researched it.

**Comment by chase-miller** [comment]
> The simplest fix right now is to get the binaries from the tilt tap instead of homebrew-core
> 
> ```
> brew install tilt-dev/tap/tilt
> ```
> 
> That will use the precompiled binaries with a known good go version. There might be a way to get homebrew-core to compile tilt with a pinned go version, but haven't researched it.

Installing from tilt-dev/tap did the trick. Thank you!

**Comment by saweber** [comment]
@nicks You are correct!
```
> strings $(which tilt) | grep go1.
go1.20.6
```
`tilt` was installed using homebrew.

Uninstalling tilt, then reinstalling with `brew install tilt-dev/tap/tilt` did the trick!

Thanks for the fast response and the great explanation!

**Comment by nicks** [comment]
should be fixed in v0.33.3, which is currently in the merge queue - https://github.com/Homebrew/homebrew-core/pull/137560
